src/gz/gz_col_view.c

```diff
@@ -1038,3 +1051,77 @@ void gz_hit_view(void)
     gz.hit_view_state = HITVIEW_INACTIVE;
   }
 }
+
+static void do_path_list(Gfx **p_gfx_p, Gfx **p_gfx_d,
+                           z64_path_t *path_list)
+{
+  const float path_marker_radius = 18.0f;
+
+  z64_path_t *cur_path = path_list;
+
+  if (cur_path != NULL && (settings->bits.path_view_points || settings->bits.path_view_lines)) {
+    while (((cur_path->points << 4) >> 28) == Z64_SEG_SCENE && cur_path->numpoints != 0) {
+      z64_xyz_t *points = (z64_xyz_t*)((z64_stab.seg[Z64_SEG_SCENE]+0x80000000) + (cur_path->points & 0x00FFFFFF));
+      if (settings->bits.path_view_lines) {
+        load_l3dex2(p_gfx_p);
```

Comment:
Hot-swapping the microcode is an expensive operation on N64 hardware and should be done as little as possible for performance reasons. The line commands should be put in a separate display list, so that swapping the microcode only has to be done once for all lines.